tests/spacepy_testing.py

```diff
@@ -202,8 +202,9 @@ def tearDown(self):
         super().tearDown()
         import matplotlib
         import matplotlib.pyplot
-        if self.save_plots and matplotlib.pyplot.get_fignums():
-            fname = 'output_{}.png'.format('_'.join(self.id().split('.')[1:]))
-            matplotlib.pyplot.savefig(fname)
+        if matplotlib.pyplot.get_fignums():
+            if self.save_plots:
+                fname = 'output_{}.png'.format('_'.join(self.id().split('.')[1:]))
+                matplotlib.pyplot.savefig(fname)
```

Comment:
This is a simple enough change, makes sense to explicitly close the file no matter what.